packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,268 @@
+    }
+  }
+
+  /**
+   * Method to set the positioning of the menu.
+   * @public
+   */
+  public setPositioning(): void {
+    if (this.$fastController.isConnected && this._trigger && this.menuListContainer) {
+      this.cleanup = autoUpdate(this, this.menuListContainer, async () => {
+        const { middlewareData, x, y } = await computePosition(this._trigger!, this.menuListContainer!, {
+          placement: 'bottom',
+          strategy: 'fixed',
+          middleware: [
+            flip(),
+            size({
+              apply: ({ availableHeight, rects }) => {
+                this.menuListContainer?.style &&
+                  Object.assign(this.menuListContainer.style, {
+                    maxHeight: `${availableHeight}px`,
+                    width: `${rects.reference.width}px`,
+                  });
+              },
+            }),
+            hide(),
+          ],
+        });
+        if (middlewareData.hide?.referenceHidden) {
+          this.expanded = false;
+          return;
+        }
+
+        this.menuListContainer?.style &&
+          Object.assign(this.menuListContainer.style, {
+            position: 'fixed',
+            top: '0',
+            left: '0',
+            transform: `translate(${x}px, ${y}px)`,
+          });
+      });
+    }
+  }
+
+  /**
+   * Method that gets called whenever the menu changes.
+   * @protected
+   */
+  protected menuChanged(): void {
+    if (this.$fastController.isConnected && this.menu && this.menu.length) {
+      this._menu = this.menu[0] as MenuList;
+    }
+  }
+
+  /**
+   * Method that gets called whenever the trigger changes.
+   * @protected
+   * @param {HTMLElement[] | void} prev - The previous value of the trigger.
+   * @param {HTMLElement[]} next - The new value of the trigger.
+   */
+  protected triggerChanged(prev: HTMLElement[] | void, next: HTMLElement[]): void {
+    if (prev !== next) {
+      if (Array.isArray(prev) && prev.length) {
+        this.cleanupTriggerEventListeners(prev[0]);
+      }
+
+      if (next && next.length) {
+        this._trigger = next[0];
+        this._trigger.setAttribute('aria-haspopup', 'true');
+        this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+        this._trigger.addEventListener('keydown', this.handleTriggerKeydown);
+        this._trigger.addEventListener('click', this.toggleMenu);
+
+        if (this.openOnHover) {
+          this._trigger.addEventListener('mouseover', this.toggleMenu);
+        }
+      }
+    }
+  }
+
+  /**
+   * Method that gets called whenever the expanded state changes.
+   * @protected
+   */
+  protected expandedChanged(oldValue: boolean, newValue: boolean): void {
+    if (this.$fastController.isConnected && this._trigger) {
+      this._trigger.setAttribute('aria-expanded', `${this.expanded}`);
+    }
+    Updates.enqueue(() => {
+      this.setPositioning();
+      this.focus();
+    });
+    this.$emit('expandedChanged', { expanded: newValue });
```

Comment:
Should we just have this as a change event and hand back more details? If we need to notify of changes unrelated to expanded, we can use the same event and pass back more details.